src/erlcloud_cloudformation.erl

```diff
@@ -0,0 +1,403 @@
+-module (erlcloud_cloudformation).
+
+-include_lib("erlcloud/include/erlcloud.hrl").
+-include_lib("erlcloud/include/erlcloud_aws.hrl").
+
+-define(API_VERSION, "2010-05-15").
+
+-type access_key_id() :: string().
+-type secret_access_key() :: string().
+
+-type params() :: proplists:proplist().
+-type cloudformation_list() :: proplists:proplist().
+
+%% Library initialization
+-export([
+    configure/2,
+    new/2
+]).
+
+
+%% Cloud Formation API Functions
+-export ([list_stacks_all/2,
+          list_stacks/2,
+          list_stack_resources_all/2,
+          list_stack_resources/2,
+          describe_stack_resources/2,
+          describe_stack_resource/2,
+          describe_stacks_all/2,
+          describe_stacks/2,
+          get_stack_policy/2,
+          get_template/2,
+          get_template_summary/2,
+          describe_account_limits/2,
+          describe_account_limits_all/1,
+          describe_stack_events_all/2,
+          describe_stack_events/2]).
+
+
+%%==============================================================================
+%% Library initialization
+%%==============================================================================
+
+-spec configure(access_key_id(), secret_access_key()) -> ok.
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+
+-spec new(access_key_id(), secret_access_key()) -> #aws_config{}.
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey
+    }.
+
+%%==============================================================================
+%% Cloud Formation API Functions
+%%==============================================================================
+
+-spec list_stacks_all(params(), aws_config()) -> {ok, cloudformation_list()}.
+list_stacks_all(Params, Config = #aws_config{}) ->
+    list_all(fun list_stacks/2, Params, Config, []).
+
+-spec list_stacks(params(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+list_stacks(Params, Config = #aws_config{}) ->
+
+     ExtraParams= lists:map(fun(T) ->
+            case T of
+                {stack_status_filter, N, Filter} ->
+                    {io_lib:format("StackStatusFilter.member.~p", [N]), Filter};
+
+                _ -> T
+            end
+        end, Params),
+
+    case cloudformation_request(Config, "ListStacks", ExtraParams) of
+        {ok, XmlNode} ->
+            NextToken = erlcloud_xml:get_text("/ListStacksResponse/ListStacksResult/NextToken", XmlNode, undefined),
+            [{next_token, NextToken}, extract_stack_summaries(xmerl_xpath:string("/ListStacksResponse/ListStacksResult", XmlNode))];
```

Comment:
please return it as `{ok, StacksSummaries, NextToken}` here and in all places related to pagination